### PR TITLE
[frontend/settings] Remove useless spaces in ACL

### LINKED ISSUE #802: [frontend/settings] Access control list adds useless blank spaces on save
When we push on the "save changes" button of the course settings page, blank spaces are added at the beginning of the first line and at the end of the last line of the ACL.

Here are the interesting part of the course.yaml:
1. after the initial addition:
```
registration_ac_list:
- test
- test2
```
2. after pushing once on "save changes"
```
registration_ac_list:
- ' test'
- 'test2 '
```
3. pushing once again
```
registration_ac_list:
- '  test'
- 'test2  '
```
4. after adding a new entry
```
registration_ac_list:
- '   test'
- 'test2   '
- test3
```
5. after pushing only on "save changes"
```
registration_ac_list:
- '    test'
- 'test2   '
- 'test3 '
```

### PR DESCRIPTION
**Pull request by nrybowski**
Fix #802

### CODE REVIEW

**Comment by anthonygego** [comment]
This is probably not enough to fix #802 as this does not ensure the specified items are trimmed before saving changes to file. It only ensures spaces are not added by the rendering process.

**Comment by nrybowski** [comment]
This commit indeed fixes #802 but I agree that further input sanitation is required.
I will add that in subsequent commits.

**Comment by nrybowski** [comment]
> This is probably not enough to fix #802 as this does not ensure the specified items are trimmed before saving changes to file. It only ensures spaces are not added by the rendering process.

Fixed in e79eddc47c9a9d52f4c85d5d66de6b2dfeab2696.

File: inginious/frontend/pages/course_admin/settings.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def POST_AUTH(self, courseid):  # pylint: disable=arguments-differ
                 errors.append(_('Invalid ACL value'))
             if course_content['registration_ac'] == "None":
                 course_content['registration_ac'] = None
-            course_content['registration_ac_list'] = data['registration_ac_list'].splitlines()
+            course_content['registration_ac_list'] = [re.sub(' ', '', line) for line in data['registration_ac_list'].splitlines()]
```
Comment (anthonygego) [inline inginious/frontend/pages/course_admin/settings.py:79]: ``line.strip()`` is probably more appropriate.

**Review by anthonygego — CHANGES_REQUESTED** [review]

File: inginious/frontend/pages/course_admin/settings.py (hunk 1 above)
Comment (nrybowski) [reply inginious/frontend/pages/course_admin/settings.py:79]: Do we accept entries such as `foo bar` in the ACL?

Comment (nrybowski) [reply inginious/frontend/pages/course_admin/settings.py:79]: I wasn't able to add tabs via the web interface but this is still possible via the yaml file.
I will replace the regex by `strip`.

**Comment by nrybowski** [comment]
Rebased on 15a948f60f6851f81d83b98414cf7f67ca2c4fef.

**Comment by anthonygego** [comment]
LGTM.

**Review by anthonygego — APPROVED** [review]

**Pull request merged**